internal/xds/resolver/serviceconfig.go

```diff
@@ -114,12 +114,35 @@ type route struct {
 	maxStreamDuration time.Duration
 	retryConfig       *xdsresource.RetryConfig
 	hashPolicies      []*xdsresource.HashPolicy
+	autoHostRewrite   bool
 }
 
 func (r route) String() string {
 	return fmt.Sprintf("%s -> { clusters: %v, maxStreamDuration: %v }", r.m.String(), r.clusters, r.maxStreamDuration)
 }
 
+// autoHostRewriteKey is the context key used to store the value of
+// route's autoHostRewrite in the RPC context.
+type autoHostRewriteKey struct{}
+
+// getMatchedAutoHostRewrite retrieves the autoHostRewrite value from the provided context.
+func getMatchedAutoHostRewrite(ctx context.Context) bool {
```

Comment:
Once this is moved to the `clusterimpl` policy, please remove the `get` prefix. See: go/go-style/decisions#getters